test/battle/hold_effect/micle_berry.c

```diff
@@ -64,3 +64,36 @@ SINGLE_BATTLE_TEST("Micle Berry raises the holder's accuracy by 1.2")
         ANIMATION(ANIM_TYPE_MOVE, MOVE_SUBMISSION, player);
     }
 }
+
+SINGLE_BATTLE_TEST("Micle Berry increases the accuracy of the next used move")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_ROCK_SLIDE].accuracy == 90);
+        PASSES_RANDOMLY(100, 100, RNG_ACCURACY);
+        PLAYER(SPECIES_WOBBUFFET) { MaxHP(100); HP(26); Item(ITEM_MICLE_BERRY); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_TACKLE); }
+        TURN { MOVE(player, MOVE_ROCK_SLIDE); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, opponent);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROCK_SLIDE, player);
+    }
+}
+
+SINGLE_BATTLE_TEST("Miscle Berry increases the accuracy of the next used move")
```

Comment:
There's a typo in the berry name, should be Micle. This would mean the two tests have identical names, so I'd suggest either going with the name I was using for the other test ("Micle Berry increases the accuracy of the next used move across turns")or coming up with something else that's representative :)

Amusingly that typo was also in the test I commented whoops I read the berry name wrong lol

Reply:
pushed a suggestion